Sources/ExchangeSdk/Network/ExchangeNetworkService.swift

```diff
@@ -0,0 +1,94 @@
+    
+    func request<T: Decodable>(with target: BaseTarget) async -> Result<T, ExchangeInchError> {
+        let asyncRequestWrapper = AsyncMoyaRequestWrapper<T> { [weak self] continuation in
+            guard let self = self else { return nil }
+            
+            return self.provider.request(target) { result in
+                switch result {
+                case .success(let response):
+                    if self.debugMode {
+                        print("URL REQUEST -> \(response.request?.url?.absoluteString ?? "")")
+                        self.responseDecoding(data: response.data, decodeTo: T.self)
+                    }
+                    
+                    if let response = try? response.filterSuccessfulStatusCodes() {
+                        do {
+                            let object = try self.jsonDecoder.decode(T.self, from: response.data)
+                            continuation.resume(returning: .success(object))
+                        } catch {
+                            continuation.resume(returning: .failure(.decodeError(error: error)))
+                        }
+                    } else {
+                        do {
+                            let errorObject = try self.jsonDecoder.decode(InchError.self, from: response.data)
+                            continuation.resume(returning: .failure(.parsedError(withInfo: errorObject)))
+                        } catch {
+                            continuation.resume(returning: .failure(.unknownError(statusCode: response.statusCode)))
+                        }
+                    }
+                case .failure(let error):
+                    if self.debugMode {
+                        print("URL REQUEST -> \(error.response?.request?.url?.absoluteString ?? "")")
+                    }
+                    continuation.resume(returning: .failure(.serverError(withError: error)))
+                }
+            }
+        }
+        
+        return await withTaskCancellationHandler(handler: {
+            asyncRequestWrapper.cancel()
+        }, operation: {
+            await withCheckedContinuation({ continuation in
+                asyncRequestWrapper.perform(continuation: continuation)
+            })
+        })
+    }
+    
+    // MARK: - Private
+    
+    private func responseDecoding<T: Decodable>(data: Data, decodeTo: T.Type) {
+        do {
+            let decodeJSON = try JSONSerialization.jsonObject(with: data, options: .mutableContainers)
+            let dataJSON = try JSONSerialization.data(withJSONObject: decodeJSON, options: .prettyPrinted)
+            print(String(decoding: dataJSON, as: UTF8.self))
+        } catch {
+            print("Decoding response error -> \(error)")
+        }
+        
+        do {
+            _ = try jsonDecoder.decode(decodeTo, from: data)
+            print("Decode to object success")
+        } catch {
+            print("Decode to object error -> \(error)")
+        }
+    }
```

Comment:
думаю стоит переделать эту функцию и вызывать её для логирования парсинга
```suggestion
    private func logIfNeeded<T: Decodable>(data: Data, decodeTo: T.Type) {
        guard isDebug else { return }
        
        do {
            let decodeJSON = try JSONSerialization.jsonObject(with: data, options: .mutableContainers)
            let dataJSON = try JSONSerialization.data(withJSONObject: decodeJSON, options: .prettyPrinted)
            print(String(decoding: dataJSON, as: UTF8.self))
        } catch {
            print("Decoding response error -> \(error)")
        }
    }
```

Reply:
✅